portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,68 @@
 
 package com.google.sps.servlets;
 
+import com.google.sps.servlets.Comment;
+import com.google.gson.Gson;
+import java.util.ArrayList;
+import java.util.List;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that handles comments data. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private List<Comment> comments = new ArrayList<Comment>();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+
+    // Convert the comments into JSON.
+    String json = convertToJson(comments);
```

Comment:
I think you can inline this.

BTW, Variables are typically declared just before they are first used (to minimize scope).
go/java-style#s4.8.2.2-variables-limited-scope
go/ej3e-57

Reply:
Done. I see, thanks for the references - I have inlined this.